std/bigint.d

```diff
@@ -78,30 +79,37 @@ public:
 
         data = 0UL;
 
+        // auto decoding special case
+        static if (isNarrowString!Range)
+            auto codeUnits = s.byCodeUnit();
+        else
+            alias codeUnits = s;
+
         // check for signs and if the string is a hex value
-        if (s.front == '+')
+        if (codeUnits.front == '+')
         {
-            s.popFront(); // skip '+'
+            codeUnits.popFront(); // skip '+'
         }
-        else if (s.front == '-')
+        else if (codeUnits.front == '-')
         {
             neg = true;
-            s.popFront();
+            codeUnits.popFront();
         }
 
-        if (s.save.startsWith("0x") || s.save.startsWith("0X"))
+        if (codeUnits.save.startsWith("0x".byUTF!(ElementEncodingType!(typeof(codeUnits)))) ||
+            codeUnits.save.startsWith("0X".byUTF!(ElementEncodingType!(typeof(codeUnits)))))
```

Comment:
Can we get away with `"0x".byChar` here? `char` converts implicitly to `wchar` and `dchar`, and the conversion is safe for ASCII. Alternatively, could cast the string literals to `immutable(ElementEncodingType!(typeof(codeUnits)))[]`.